### DISCUSSION TITLE
Using router hardware (OpenWRT) & POE with pymagnum

### DISCUSSION THREAD

**Discussion by Loafdude** [CharlesGodwin/pymagnum#32] [Show and tell]
For anyone interested, it is very possible to run pymagnum on a router with USB.

Min router requirements;
- OpenWRT support
- 16+MB flash 
- Has USB port
- I successfully used a GL-iNet GL-MT300N-V2

You will also need;
- USB to RS485 hardware (This can be USB -> RS485 or USB->RS232->RS485 Adapter)
- Optional POE adapter to extract 5v. I used a USB0503G. I allows for 3A@5V which is plenty. I am unsure of exact power consumption. 

OpenWRT issues to overcome
- Flash is severely limited. 
- Due to this pip cannot be installed so packages need manual installation
- Install python3-light and python3 mqtt and python serial packages
- Install kmod usb for your USB serial adapter type
- scp or wget a copy of pymagnum, pytz_deprecation_shim and tzlocal pypi packages to the /tmp/ directory
- Copy the magnum module directory into /usr/pymagnum & copy mqttlogger.py from tools into /usr/pymagnum
- Manually install pytz_deprecation_shim into /usr/lib/python3.9/site-packages
- Copy tzlocal directory into /usr/pymagnum

This is a quick script to keep mqttlogger.py running with a 5 second timeout
/etc/init.d/ups
```
#!/bin/sh /etc/rc.common
until /usr/bin/python /usr/pymagnum/mqttlogger.py --broker 10.9.8.184 --topic Home/UPS/; do
    echo "mqttlogger.py crashed with exit code $?.  Respawning in 5 sec.." >&2
    sleep 5
done
```
you will need to symlink the script into rc.d
```
ln -s /etc/init.d/ups /etc/rc.d/S99ups
```

I likely missed something. Feel free to ask questions here

![image](https://user-images.githubusercontent.com/17401709/158316603-84bc5d9b-886d-47fd-abce-2d58a07a990e.png)
![4032-3024-max](https://user-images.githubusercontent.com/17401709/158316746-a854c7b1-2399-4a2b-a659-7abe4a512ace.jpg)

**Comment by CharlesGodwin** [comment]
Thanks for your contribution. This looks good but I must add that I have NO experience with OpenWRT.
